### ISSUE TITLE
Force display of iterations per second (it/s) - instead of displaying inverse (s/it) based on rate

### ISSUE THREAD

**Issue by tloki** [tqdm/tqdm#1259]
For a given code example:

```
from tqdm import tqdm
from time import sleep

for i in tqdm(range(100)):
    sleep(1.5)
```

the output looks like this:
`6%|▌         | 6/100 [00:09<02:21,  1.50s/it]`

but I'd like for it to actually force display it/s (generally, units per time, and possibly vice versa) so the output for this example (with the required parameter being set) would look like this:

`6%|▌         | 6/100 [00:09<02:21,  0.67it/s]`

as I see it, this functionality is basically hardcoded in std.py (in master branch as of October 2021) https://github.com/tqdm/tqdm/blob/fc69d5dcf578f7c7986fa76841a6b793f813df35/tqdm/std.py#L450

a modification of the above code line to
`rate_fmt = rate_noinv_fmt`

achieves desired behavior, but does not parametrize the rate when each format should be displayed. As I see it right now, that hardcoded "1" may be the only thing in need of parametrizing (possibly member variable) with an option to be set to None or so to completely ignore it, while a default value could be 1 to preserve current default behavior.

is there any other way right now to "force" set it/s mode (without changing tqdm code), and possibly set the "trigger" frequency? Say, for example, one might want to display seconds per iteration only if rate is lower than 1e-3 it/s)...

EDIT 1, 2 & 3: sentence formatting, spelling, punctuation

**Comment by alexreg** [comment]
Indeed, this is clearly undesirable in the case your rate hovers around 1, in which case you can get the unit flipping between itself and its inverse quite frequently!

**Comment by roflmaostc** [comment]
Ha!

Got fooled by this, this is really nasty imho.

**Comment by aalanwyr** [comment]
I also hit this confusion of the tqdm unit (it/s and s/it)
Luckily， we can use a tag to fix this unit :   bar_format="{l_bar}{bar}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_noinv_fmt}]"
Such as:
> for rep in tqdm(range(10), desc="Inference", ncols=100, total=10, leave=True):
>     sleep(2)
>     print("iteration:", rep)
> 
> Inference:  10%|█████▎                                               | 1/10 [00:02<00:18,  2.00s/it]iteration: 1
> 
> for rep in tqdm(range(10), desc="Inference", ncols=100, total=10, leave=True, bar_format="{l_bar}{bar}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_noinv_fmt}]"):
>     sleep(2)
>     print("iteration:", rep)
> 
> Inference:  10%|█████▎                                               | 1/10 [00:02<00:18,  0.50it/s]iteration: 1

**Comment by clemisch** [comment]
So there is no built-in switch for this? This explicit bar format seems very verbose.

**Comment by kylematoba** [comment]
Would also like to see some sort of backwards-compatible flag that could force iteration / time, time / iteration, or the current behavior.
